codegen/build.gradle.kts

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+plugins {
+    `java-library`
+    `maven-publish`
+    signing
+    checkstyle
+    jacoco
+    id("com.github.spotbugs") version "1.6.10"
+    id("io.codearte.nexus-staging") version "0.21.0"
+}
+
+allprojects {
+    group = "software.amazon.smithy"
+    version = "0.1.0"
+}
+
+// The root project doesn't produce a JAR.
+tasks["jar"].enabled = false
+
+// Load the Sonatype user/password for use in publishing tasks.
```

Comment:
Is only the `smithy-aws-typescript-codegen` portion going to Maven Central for now?

Reply:
Yes. We don't have a need to publish the models and client generation package.